Config/SEGGER_RTT_Conf.h

```diff
@@ -152,10 +152,10 @@ Revision: $Rev: 24316 $
 #if ((defined(__SES_ARM) || defined(__SES_RISCV) || defined(__CROSSWORKS_ARM) || defined(__GNUC__) || defined(__clang__)) && !defined (__CC_ARM) && !defined(WIN32))
   #if defined(__ZEPHYR__) && defined (CONFIG_SEGGER_RTT_CUSTOM_LOCKING)
     #ifdef CONFIG_MULTITHREADING
-      extern void zephyr_rtt_mutex_lock(void);
-      extern void zephyr_rtt_mutex_unlock(void);
-      #define SEGGER_RTT_LOCK() zephyr_rtt_mutex_lock()
-      #define SEGGER_RTT_UNLOCK() zephyr_rtt_mutex_unlock()
+      extern void zephyr_rtt_spin_lock(void);
```

Comment:
I'm afraid that this can result in long periods when interrupts are locked since it blocks for RTT API where you write data to it and one can write 10 bytes to RTT is single call but you can also write 1k of data in a single call. In case of logging RTT backend, it writes from lowest priority thread and amount of data depends on internal buffer size which is configurable. By default it's 16 bytes (which result in 4,5us lock time @64MHz). When I increase this buffer to 64 bytes (which is reasonable value) lock time increased to 10us.